test/io/jesi/backpack/collection_test.cljc

```diff
@@ -871,6 +871,50 @@
       {:a 1}
       {:a 1, :b 2})))
 
+(let [one {:id 1 :i 2 :s "a" :b true}
+      two {:id 2 :i 4 :s "b" :b true}
+      three {:id 3 :i 6 :s "c" :b false}
+      coll [one two three]]
+  (deftest filter-by-test
+
+      (testing "returns empty seq when filter doesn't match anything"
+        (are [key pred] (is= [] (filter-by key pred coll))
+          :i (bp/partial-right < 1)
+          :i (bp/partial-right < 1)
+          :i (bp/p= 0)
+          :id (bp/p= 5)
+          :id (constantly false)
+          :s (bp/p= "woah")))
+
+      ; matches clojure.core/filter
+      (testing "returns empty seq when coll is nil"
+        (is= () (filter-by :id (constantly true) nil)))
+
+      (testing "can return 1 result"
+        (is= [one] (filter-by :id (bp/p= 1) coll)))
+
+      (testing "can return multiple results"
+        (is= [two three] (filter-by :id (bp/partial-right > 1) coll))
```

Comment:
```suggestion
        (is= [two three] (filter-by :id (partial < 1) coll))
```

Our convention is to always try and use `<` where possible.

Reply:
I found this confusing when you are finding the items which are > 1.
I might just use a different predicate.

Reply:
changed